### DISCUSSION TITLE
Using replicaCount: 2 in Helm chart sends all alerts twice

### DISCUSSION THREAD

**Discussion by tsboris** [jertel/elastalert2#451] [Q&A] [answered]
Hi,

I've deployed the 2.2.1 Helm chart in an EKS and I've changed replicaCount from 1 to 2 (replicaCount: 2) for HA.

After configuring some alerts, I've noticed that each alert is sent twice for a single query hit.
I've reduced the number of replicas to one and now the alerts are been sent once per query hit.
I've increased the replicaCount back to 2 and again, all alerts are send twice for a single query hit.

Is there a setting I need to set to be using a replicaCount > 1?

Thanks :-)

**Comment by ferozsalam** [comment, marked as answer]
I will defer to @jertel's opinion on this, but my understanding of the way ElastAlert works suggests that it does not currently support multiple replicas working in tandem.

ElastAlert simply runs in a loop, with an instance maintaining a queue of rules to be run and writing its current search status back to an index within Elasticsearch. Multiple replicas would have no knowledge of each other and would therefore have no way of preventing race conditions. 

A [similar question](https://github.com/Yelp/elastalert/issues/478) from some time ago with a reply from the original chief contributor suggests that a single instance is all that is supported, with no concept of coordination between instances.
